Makefile.am

```diff
@@ -189,6 +190,7 @@ tools_tpm2_pcrextend_SOURCES = tools/tpm2_pcrextend.c $(TOOL_SRC)
 tools_tpm2_pcrevent_SOURCES = tools/tpm2_pcrevent.c $(TOOL_SRC)
 tools_tpm2_rc_decode_SOURCES = tools/tpm2_rc_decode.c $(TOOL_SRC)
 tools_tpm2_import_SOURCES = tools/tpm2_import.c $(TOOL_SRC)
+tools_tpm2_print_quote_SOURCES = tools/tpm2_print_quote.c
```

Comment:
I think this would be a good time to start tools/aux where we can place this tool and move tpm2_rc_decode and tpm2_akparse. As well as make it a configure option.

Reply:
Also this needs a manpage

Reply:
I like the idea of auxiliary tools, which don't directly interact with TPM. I'll move some things around.

I'm not sure what you mean by "configure option". Are you saying aux tools should optionally build?

Reply:
yes aux tools id see as being optionally built. Maybe opt in be default and allow opt out. @martinezjavier thoughts?

Reply:
@williamcroberts I wonder what would be the value of not building them. Are these auxiliary tools (tpm2_akparse, tpm2_print_quote) meant to be needed for TPM2 use cases or are mostly for debugging problems like tpm2_rc_decode? If the latter, I wonder if tools/debug shouldn't be a better name.

Also, this sub-directory will only have some meaning when executing the built tools in the source code directory, right? Since regardless of their path, all will be part of bin_PROGRAMS and will end in the bindir when are installed with make install.

So maybe it will be confusing for users to only have a sub-set of the existing tpm2 tools in bindir if the configure option is disabled.

Reply:
@martinezjavier I see aux tools as tools that don't invoke tpm commands. They wouldn't be needed, but mostly for conversion routines, debugging, etc. I don't want to bloat the core tool set with things that are not needed all the time.

I also don't package things for distro use, so I wasn't sure if there is any convention/wisdom you might have in this area.

I could see how some packages having extra tools could be confusing.

So is our consensus, since we don't have a very strong opinion just adding them directly into the build for now?

Reply:
@williamcroberts I see. What many packages do to have a core package and another package that has additional stuff (usually the convention is to use a -extras suffix, i.e: tpm2-tools-extras).

But I guess it will depend whether people will really be interested in doing this split or it's most likely that you will either want all the tools (the ones using TPM2 commands and the auxiliary ones) or none.

So yeah, I think for now we can just keep all of them in the build. Although I don't really have a strong opinion.

Somehow related to this, we should allow tools that don't use the SAPI context to not fail if the context can't be created (in fact I think it shouldn't even be attempted to create a context for these). For example, this shouldn't really fail:

```
$ tpm2_rc_decode 0x18a
ERROR: Failed to initialize tcti context: 0x1
```